stac_generator/raster_data/generator.py

```diff
@@ -0,0 +1,171 @@
+import csv
+import json
+import logging
+from datetime import UTC
+from typing import Any
+
+import pandas as pd
+import pystac
+import rasterio
+from pystac.extensions.eo import Band, EOExtension
+from pystac.extensions.projection import ProjectionExtension
+from pystac.extensions.raster import AssetRasterExtension, DataType, RasterBand
+from shapely.geometry import Polygon, mapping
+
+from stac_generator.base.generator import StacGenerator
+from stac_generator.base.schema import StacCatalogConfig, StacCollectionConfig
+
+from .schema import RasterSourceConfig
+
+# In case the config file has different names for bands
+BAND_MAPPING: dict[str, str] = {
+    "red": "red",
+    "green": "green",
+    "blue": "blue",
+    "nir": "nir",
+    "rededge": "rededge",
+    "ndvi": "ndvi",
+    "ndvi2": "ndvi2",
+}
+
+
+class RasterGenerator(StacGenerator[RasterSourceConfig]):
+    source_type = RasterSourceConfig
+
+    def __init__(
+        self,
+        source_df: pd.DataFrame,
+        collection_cfg: StacCollectionConfig,
+        catalog_cfg: StacCatalogConfig | None = None,
+        href: str | None = None,
+    ) -> None:
+        # Reset the DataFrame index
+        source_df = source_df.reset_index(drop=True)
+
+        # Ensure the EPSG column is properly formatted
+        if "epsg" in source_df.columns:
```

Comment:
You shouldn't need to redo type validation. Pydantic should have already handled this

Reply:
removed